### PR TITLE
Ampers - Sara S

### PR DESCRIPTION
**Pull request by CheerOnMars**
# JS Scrabble
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | The patterns I've noticed most were adding ||'s in my conditional statements, and adding +='s in my loops.
What was a challenge you faced in this assignment? | Besides keeping track of () and {}, it took some thinking through to get my head around 'this.'
Do you have any recommendations on how we could improve this project for the next cohort? |

### CODE REVIEW

File: scrabble.js [hunk 1]
```diff
@@ -1,16 +1,124 @@
...
+};
+
 
 const Scrabble = {
   score(word) {
+    let total = 0;
+    let letters = word.toUpperCase().split(``);
+
+    if (word.match(/^[a-zA-Z]{1,7}$/) == null) {
+      throw `How'd you get a non-letter tile, even?`;
```
Comment (CheezItMan) [inline scrabble.js:37]: :)

File: scrabble.js [hunk 2]
```diff
@@ -1,16 +1,124 @@
...
+  totalScore() {
+    let total = 0;
+    for (let word of this.plays) {
+      total += Scrabble.score(word);
+    }
+    return total;
+  }
+
+  hasWon() {
+    if (this.totalScore() >= 100) {
```
Comment (CheezItMan) [inline scrabble.js:106]: You could also simply 
```JavaScript
return this.totalScore() >= 100;
```

File: specs/scrabble.spec.js [hunk 3]
```diff
@@ -181,45 +181,55 @@ describe('Player', () => {
   });
 
   describe('hasWon', () => {
-    test.skip('returns false when score < 100', () => {
-
-
+    test('returns false when score < 100', () => {
+      const player = new Scrabble.Player('test player');
```
Comment (CheezItMan) [inline specs/scrabble.spec.js:185]: It would be good to loop through until you reach 99, the edge case, and verify that `hasWon()` is `false`.

File: specs/scrabble.spec.js [hunk 4]
```diff
@@ -181,45 +181,55 @@ describe('Player', () => {
...
+      player.play('qqqqxaa');
+      expect(player.totalScore()).toBe(100);
+      expect(player.hasWon()).toBe(true);
     });
 
-    test.skip('returns true when score > 100', () => {
-
-
+    test('returns true when score > 100', () => {
+      const player = new Scrabble.Player('test player');
```
Comment (CheezItMan) [inline specs/scrabble.spec.js:199]: It would also be good to verify that 101 makes `hasWon()` true

**Comment by CheezItMan** [comment]
# JS Scrabble
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** | 
Git hygiene | You should have more commits.  The commit messages are good.
Comprehension questions | Check, it' normal to have issues with `this`.
**General** | 
`score` calculates score, has appropriate params and return value | Check
`highestScoreFrom` calculates highest scoring word, has appropriate params and return value | Check
**`Player` object** | 
Has `name` and `plays` properties | Check
Has `play`, `totalScore`, `hasWon` functions | Check
Has `highestScoringWord` and `highestWordScore` functions | Check
**Overall** | Well done you hit all the learning goals, well done!
